From ca14a9f02b763d07e07e92465b5dccb7366c9ed1 Mon Sep 17 00:00:00 2001
From: Castix <vincenzo.castiglia@axant.it>
Date: Tue, 15 Dec 2020 13:35:33 +0100
Subject: [PATCH 04/16] fix line too long in tests

---
 gjwt/tests/test_jwk.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gjwt/tests/test_jwk.py b/gjwt/tests/test_jwk.py
index f2e81fa..c28a610 100644
--- a/gjwt/tests/test_jwk.py
+++ b/gjwt/tests/test_jwk.py
@@ -73,7 +73,7 @@ def test_jwk_from_dict_unsupported_kty():
 def test_jwk_from_bytes_argument_conversion_confusing_name():
     with raises(Exception) as ex:
         @jwk_from_bytes_argument_conversion
-        def confusing(_content, *, _password=None, _backend=None):  # pylint: disable=unused-variable
+        def confusing():  # pylint: disable=unused-variable
             pass
     assert ("the wrapped function must have either public"
             " or private in it's name" in str(ex))
-- 
2.25.1

