From de579f5bb5637fd2174212473e9d4164232d3408 Mon Sep 17 00:00:00 2001
From: Castix <vincenzo.castiglia@axant.it>
Date: Thu, 7 Jan 2021 14:40:45 +0100
Subject: [PATCH 10/16] use functools.wraps to keep doctring in jwk

---
 gjwt/jwk.py | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/gjwt/jwk.py b/gjwt/jwk.py
index 312aca0..f3c42c1 100644
--- a/gjwt/jwk.py
+++ b/gjwt/jwk.py
@@ -26,6 +26,7 @@ from typing import (
     Union,
     Optional
 )
+from functools import wraps
 from cryptography.exceptions import InvalidSignature
 from cryptography.hazmat.backends import default_backend
 from cryptography.hazmat.primitives.asymmetric import padding
@@ -285,6 +286,7 @@ def jwk_from_bytes_argument_conversion(func):
         raise Exception("the wrapped function must have either public"
                         " or private in it's name")
 
+    @wraps(func)
     def wrapper(content, **kwargs):
         # content is the only positional argument allowed
         # the function you wrap must use kwargs-only arguments PEP3102
-- 
2.25.1

